src/pages/register-page/constants.ts

```diff
@@ -1,57 +1,164 @@
-import { messages } from './messages.ts';
+import type { FormField, FormValues } from './types.ts';
+import { getCountryOptions } from '../../components/country-select/countries.ts';
+import type { RegisterOptions } from 'react-hook-form';
+import {
+  postcodeValidator,
+  postcodeValidatorExistsForCountry,
+} from 'postcode-validator';
 
-export const FIELDS = [
+const countryOptions = getCountryOptions();
+
+export const FIELDS: FormField[] = [
   {
     type: 'email',
-    label: messages.fields.email.label,
-    placeholder: messages.fields.email.placeholder,
-    name: messages.fields.email.name,
+    label: 'Email',
+    placeholder: 'Email',
+    name: 'email',
   },
   {
     type: 'password',
-    label: messages.fields.password.label,
-    placeholder: messages.fields.password.placeholder,
-    name: messages.fields.password.name,
+    label: 'Password',
+    placeholder: 'Password',
+    name: 'password',
   },
   {
     type: 'text',
-    label: messages.fields.firstName.label,
-    placeholder: messages.fields.firstName.placeholder,
-    name: messages.fields.firstName.name,
+    label: 'First Name',
+    placeholder: 'First Name',
+    name: 'firstName',
   },
   {
     type: 'text',
-    label: messages.fields.lastName.label,
-    placeholder: messages.fields.lastName.placeholder,
-    name: messages.fields.lastName.name,
+    label: 'Last Name',
+    placeholder: 'Last Name',
+    name: 'lastName',
   },
   {
     type: 'date',
-    label: messages.fields.birth.label,
-    placeholder: messages.fields.birth.placeholder,
-    name: messages.fields.birth.name,
+    label: 'Birth',
+    placeholder: 'Birth',
+    name: 'birth',
   },
   {
     type: 'country-select',
-    label: messages.fields.country.label,
-    name: messages.fields.country.name,
+    label: 'Country',
+    name: 'country',
   },
   {
     type: 'text',
-    label: messages.fields.city.label,
-    placeholder: messages.fields.city.placeholder,
-    name: messages.fields.city.name,
+    label: 'City',
+    placeholder: 'City',
+    name: 'city',
   },
   {
     type: 'text',
-    label: messages.fields.street.label,
-    placeholder: messages.fields.street.placeholder,
-    name: messages.fields.street.name,
+    label: 'Street',
+    placeholder: 'Street',
+    name: 'street',
   },
   {
     type: 'text',
-    label: messages.fields.postCode.label,
-    placeholder: messages.fields.postCode.placeholder,
-    name: messages.fields.postCode.name,
+    label: 'Post code',
+    placeholder: 'Post code',
+    name: 'postCode',
   },
 ];
+
+export const validationRules: Record<
+  keyof FormValues,
+  RegisterOptions<FormValues>
+> = {
+  email: {
+    required: 'Email is required',
+    pattern: {
+      value: /^\S+@\S+\.\S+$/,
+      message: 'Invalid email address',
+    },
+  },
+  password: {
+    required: 'Password is required',
+    minLength: {
+      value: 8,
+      message: 'Password must be at least 8 characters',
```

Comment:
давай валидационные сообщения оставим как есть, без месаг. в целом, валидационные сообщения или сообщения об ошибках допускается так держать. но все значения надо вынести в константы а именно


```suggestion
  email: {
    required: 'Email is required',
    pattern: {
      value: /^\S+@\S+\.\S+$/,
      message: 'Invalid email address',
    },
  },
  password: {
    required: 'Password is required',
    minLength: {
      value: MIN_PASSWORD_LENGTH,
      message: `Password must be at least ${MIN_PASSWORD_LENGTH} characters`,
```